deduction_rules.py

```diff
@@ -167,4 +167,162 @@ def line_square(sudoku):
                     r,c=local_to_global(sec,i,j)
                     if c!=col:
                         made_deduction|=sudoku.ban(r,c,val+1,"col_square",reason)
+    return made_deduction
+
+def naked_trios(sudoku):
+    """RULE: if v,w,x can be written only to three cells in this row/col/sec, then remove it from the other cells of the row/col/sec."""
+
+    def search_for_naked_trios(elems):
+        """Returns a list of 3-tuples, where every tuple contains the indeces of the identical inputs. The cases:
+        (123) (123) (123) - {3/3/3} (in terms of candidates per cell)
+        (123) (123) (12) - {3/3/2} (or some combination thereof)
+        (123) (12) (23) - {3/2/2/}
+        (12) (23) (13) - {2/2/2}
+        """
```

Comment:
Nice example! Unforunately I have no idea what it's supposed to illustrate. `search_for_naked_trios` returns a list of 3-tuples, which contains the subsets of the list `elems` who satisfy the following property: there are only three numbers which can be written to at least one of the cells in the trio? Seems to be the case.

This also conflicts the doctrsing of `naked_trios`, but conforms the original definition.